packaging/macos/lib/installer_lib.sh

```
@@ -200,154 +212,417 @@ _native_claudecode_version_from_dir() {
+  local a="$1" b="$2"
+  # Fast path: identical strings are not "greater than".
+  [[ "${a}" == "${b}" ]] && return 1
+  # Split into <main>[-<pre>].
+  local a_main="${a%%-*}" a_pre=""
+  local b_main="${b%%-*}" b_pre=""
+  if [[ "${a}" == *-* ]]; then a_pre="${a#*-}"; fi
+  if [[ "${b}" == *-* ]]; then b_pre="${b#*-}"; fi
+  # Compare MAJOR.MINOR.PATCH numerically. Extra numeric identifiers
+  # past the third position (rare in practice) also participate.
+  local IFS=.
+  local -a am=(${a_main}) bm=(${b_main})
```

Shellcheck (0.11.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)